src/main/java/uk/gov/hmcts/ccd/domain/service/message/CaseEventMessageService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.ccd.domain.service.message;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.casedetails.CaseAuditEventRepository;
+import uk.gov.hmcts.ccd.data.message.MessageCandidateRepository;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.std.MessageInformation;
+import uk.gov.hmcts.ccd.domain.model.std.MessageQueueCandidate;
+
+import javax.inject.Inject;
+import java.time.Clock;
+
+@Service
+@Qualifier("caseEventMessageService")
+public class CaseEventMessageService extends AbstractMessageService {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private final MessageCandidateRepository messageCandidateRepository;
+
+    @Inject
+    public CaseEventMessageService(@Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                                   final MessageCandidateRepository messageCandidateRepository,
+                                   CaseAuditEventRepository caseAuditEventRepository,
+                                   @Qualifier("utcClock") final Clock clock) {
+        super(userRepository, caseAuditEventRepository, clock);
+        this.messageCandidateRepository = messageCandidateRepository;
+    }
+
+    @Override
+    public void handleMessage(MessageContext messageContext) {
+        final MessageQueueCandidate messageQueueCandidate = new MessageQueueCandidate();
+        if (Boolean.TRUE.equals(messageContext.getCaseEventDefinition().getPublish())) {
+
+            MessageInformation messageInformation = populateMessageInformation(messageContext);
+            JsonNode node = mapper.convertValue(messageInformation, JsonNode.class);
+
+            messageQueueCandidate.setMessageInformation(node);
+            messageQueueCandidate.setMessageType(MessageType.CASE_EVENT.name());
+            messageQueueCandidate.setTimeStamp(now());
```

Comment:
⚠️ wondering whether to avoid race conditions we should let the DB set this value when the row is inserted in the table (I think the column is already configured to have a default). 
With this implementation what can happen is that the Java code sets a certain value for the timestamp. But the thread can be put to sleep and resume only after some time later. Therefore the timestamp does not correctly reflect the message creation time and this could cause problems to the message publisher process